hs/CalendarResponse.hs

```diff
@@ -0,0 +1,281 @@
+module CalendarResponse where
+
+import Data.List (sort)
+import Data.List.Split (splitOn)
+import Data.Time (Day, UTCTime, addDays, formatTime, getCurrentTime)
+import Happstack.Server (ServerPart, Response, toResponse)
+import Control.Monad.IO.Class (liftIO)
+import System.Locale
+import Database.CourseQueries (returnTutorialTimes, returnLectureTimes)
+import qualified Data.Text as T
+import Database.Tables as Tables
+import Config (firstMondayFall,
+               lastWednesdayFall,
+               firstMondayWinter,
+               lastMondayWinter,
+               outDay)
+
+-- | Returns an ICS file of events as requested by the user.
+calendarResponse :: String -> ServerPart Response
+calendarResponse courses =
+    liftIO $ getCalendar courses
+
+-- | Gets together all the pieces of the program.
+getCalendar :: String -> IO Response
+getCalendar courses = do
+    let courseInfo = getCoursesInfo courses
+    databaseInfo <- mapM pullDatabase courseInfo
+    currentTime <- getCurrentTime
+    let timeSystem = getTime currentTime
+    let events = databaseInfo >>= (getEvent timeSystem)
+    return $ toResponse $ getICS events
+
+-- | Generates a properly formatted current date and time.
+getTime :: UTCTime -> String
+getTime currentTime = formatTime defaultTimeLocale "%Y%m%dT%H%M%SZ" currentTime
+
+-- | Generates a string representing a ICS file.
+getICS :: [String] -> String
+getICS events =
+    if null events
+    then ""
+    else unlines $ header ++ events ++ bottom
+    where
+        header = ["BEGIN:VCALENDAR", "VERSION:2.0",
+                  "PRODID:-//Courseography//Calendar",
+                  "CALSCALE:GREGORIAN", "METHOD:PUBLISH"]
+        bottom = ["END:VCALENDAR"]
+
+-- ** Cookie Information
+
+-- | Obtains a list with all the information about the courses obtained from 
+-- the cookies.
+getCoursesInfo :: String -> [(String, String, String)]
+getCoursesInfo courses = map courseInfo allCourses
+    where
+        courseInfo course = (course !! 0, course !! 1, course !! 2)
+        allCourses = map (splitOn "-") byCourse
+        byCourse = splitOn "_" courses
+
+-- ** Database Information
+
+-- | Pulls out the information for each course from the database.
+pullDatabase :: (String, String, String) -> IO ([Time], String, String, String)
+pullDatabase (code, sect, session) =
+    if take 1 sect == "L"
```

Comment:
You can use pattern matching here. I believe @david-yz-liu suggested this before.